### ISSUE TITLE
[Bug]:  bigquery_tools.parse_table_reference accepts invalid identifier

### ISSUE THREAD

**Issue by julianotoriani** [apache/beam#32611]
### What happened?

While doing some tests, `WriteToBigQuery` started returning an error (real values replaced):

>HttpError accessing <https://bigquery.googleapis.com/bigquery/v2/projects/project%3Adataset/jobs?alt=json>: response: <{\'content-type\': \'application/json; charset=UTF-8\', \'date\': \'Mon, 30 Sep 2024 14:56:38 GMT\', \'alt-svc\': \'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000\', \'transfer-encoding\': \'chunked\', \'status\': \'400\'}>, content <{"error":{"code":400,"message":"ProjectId must be non-empty","errors":[{"message":"ProjectId must be non-empty","domain":"global","reason":"badRequest"}],"status":"INVALID_ARGUMENT"}}

I noticed the dataset name being included in the project identifier i.e: "/v2/projects/**project:dataset**/jobs"

Due to a mistake on my side, the table ref was being passed as `project:dataset.dataset.table`, which `bigquerytools.parse_table_reference`  seems happy to accept:

    >>> bigquery_tools.parse_table_reference("project:dataset.dataset.table")
    <TableReference
     datasetId: 'dataset'
     projectId: 'project:dataset'
     tableId: 'table'>

When I expected it to throw the error it usually throws for invalid identifiers:

> ValueError: Expected a table reference (PROJECT:DATASET.TABLE or DATASET.TABLE) instead of ...

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by TanuSharma2511** [comment]
.take-issue

**Comment by Abacn** [comment]
colon in GCP project id is valid. The pattern was correct one

**Comment by Abacn** [comment]
Yes both domain-scoped project and container registry (in your link) are deprecated so one won't be able to create new one, and likely this is the reason it has been removed from GCP doc, but there are existing ones
